modules/geo-layers/src/quadkey-layer/quadkey-layer.js

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2015 - 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {CompositeLayer} from '@deck.gl/core';
+import {PolygonLayer} from '@deck.gl/layers';
+
+import {createCellLayer} from '../common/cell-layer-utils';
+import {getQuadkeyPolygon} from './quadkey-utils';
+
+const defaultProps = {
+  ...PolygonLayer.defaultProps,
+  getQuadkey: {type: 'accessor', value: d => d.quadkey}
+};
+
+export default class QuadkeyLayer extends CompositeLayer {
```

Comment:
It will be easier to read if you create a generic `GeoCellLayer` that has a method `indexToBounds `. Then `S2Layer`, `QuadkeyLayer` and `H3ClusterLayer` can just extend this base layer and implement their own `indexToBounds`.

Reply:
Done